src/NuGet.Core/NuGet.PackageManagement/Projects/DefaultProjectServices.cs

```diff
@@ -0,0 +1,109 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using NuGet.Frameworks;
+using NuGet.LibraryModel;
+using NuGet.Packaging.Core;
+using NuGet.ProjectModel;
+
+namespace NuGet.ProjectManagement
+{
+    /// <summary>
+    /// Null-object with no-op implementation of project services.
+    /// </summary>
+    internal sealed class DefaultProjectServices
+        : INuGetProjectServices
+        , IProjectBuildProperties
+        , IProjectScriptHostService
+        , IProjectSystemCapabilities
+        , IProjectSystemReferencesReader
+        , IProjectSystemReferencesService
+        , IProjectSystemService
+    {
+        public static INuGetProjectServices Instance { get; } = new DefaultProjectServices();
+
+        public IProjectBuildProperties BuildProperties => this;
+        public IProjectSystemCapabilities Capabilities => this;
+        public IProjectSystemReferencesReader ReferencesReader => this;
+        public IProjectSystemService ProjectSystem => this;
+        public IProjectSystemReferencesService References => this;
+        public IProjectScriptHostService ScriptService => this;
+
+        public bool SupportsPackageReferences => false;
+
+        public Task AddOrUpdatePackageReferenceAsync(
+            LibraryDependency packageReference,
+            CancellationToken _)
+        {
+            throw new NotSupportedException();
+        }
+
+        public Task<IEnumerable<LibraryDependency>> GetPackageReferencesAsync(
+            NuGetFramework targetFramework,
+            CancellationToken _)
+        {
+            throw new NotSupportedException();
+        }
+
+        public Task<IEnumerable<ProjectRestoreReference>> GetProjectReferencesAsync(
+            Common.ILogger _,
+            CancellationToken __)
+        {
+            return Task.FromResult(Enumerable.Empty<ProjectRestoreReference>());
+        }
+
+        public string GetPropertyValue(string propertyName)
+        {
+            return null;
```

Comment:
It's not clear what the pattern is:  do methods return a default value or throw?  It seems like a mixture of the two.